frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -236,8 +241,30 @@ export const insightLogic = kea<insightLogicType>({
                     for (const key of Object.keys(metadata)) {
                         updatedInsight[key] = response[key]
                     }
+
                     savedInsightsLogic.findMounted()?.actions.loadInsights()
                     dashboardsModel.actions.updateDashboardInsight(updatedInsight)
+
+                    lemonToast.success(`Updated insight`, {
+                        button: {
+                            label: 'Undo',
+                            action: async () => {
+                                const response = await api.update(
+                                    `api/projects/${teamLogic.values.currentTeamId}/insights/${values.insight.id}`,
+                                    beforeUpdates
+                                )
+                                // only update the fields that we changed
+                                const revertedInsight = { ...values.insight } as InsightModel
+                                for (const key of Object.keys(beforeUpdates)) {
+                                    revertedInsight[key] = response[key]
+                                }
+                                savedInsightsLogic.findMounted()?.actions.loadInsights()
+                                dashboardsModel.actions.updateDashboardInsight(revertedInsight)
+                                actions.setInsight(revertedInsight, { overrideFilter: false, fromPersistentApi: true })
+                                lemonToast.success('Insight change reverted')
```

Comment:
Will you consider me a bastard if I think this should be tested 😅

Reply:
I don't see how a bug could sneak in there 🤣

Reply:
Hmmm, testing this isn't straightforward. I think this will need to be Cypress